62_CAD/DrawResourcesFiller.cpp

```diff
@@ -435,66 +440,97 @@ void DrawResourcesFiller::finalizeTextureCopies(SIntendedSubmitInfo& intendedNex
 
 	auto msdfImage = msdfTextureArray->getCreationParameters().image;
 
+	msdfTextureArrayIndicesUsed.clear();
+
+	if (!textureCopies.size())
+		return;
+
 	// preparing images for copy
-	// @Lucas TODO: add a vector<image_barrier_t> push_back a barrier like below for each of the indices in `textureCopies`
-	IGPUCommandBuffer::SPipelineBarrierDependencyInfo::image_barrier_t barriersBeforeCopy[] =
+	using image_barrier_t = IGPUCommandBuffer::SPipelineBarrierDependencyInfo::image_barrier_t;
+	std::vector<image_barrier_t> barriers;
+	barriers.reserve(textureCopies.size());
```

Comment:
remove this reserve textureCopies.size()?
you decided to do all in 1 barrier, that's ok.

Reply:
updated comment

Reply:
Done